src/main/java/seedu/duke/exercisegenerator/GenerateExercise.java

```diff
@@ -31,19 +31,31 @@ public GenerateExercise() {
     public ArrayList<ExerciseData> generateRandomSet(int count) {
         Random random = new Random();
         ArrayList<ExerciseData> filteredExerciseList = new ArrayList<>();
-        for (int i = 0; i < count; i++) {
-            ExerciseData randomExercise = exerciseData.get(random.nextInt(exerciseData.size()));
-            filteredExerciseList.add(randomExercise);
+        boolean[] isInNewList = new boolean[exerciseData.size()]; 
+
+        while (filteredExerciseList.size() != count) {
+            int randomExerciseNumber = random.nextInt(exerciseData.size());
+            if (!isInNewList[randomExerciseNumber]) {
+                ExerciseData randomExercise = exerciseData.get(randomExerciseNumber);
+                filteredExerciseList.add(randomExercise);
+                isInNewList[randomExerciseNumber] = true;
+            }
```

Comment:
If user inputs a negative count, program will be thrown into an infinite loop. Hence Java CI check too forever to complete.
Actually can jus pop ```randomExercise``` from ```exerciseData```